### PR TITLE
Updated Neo4j.jl for Julia 0.7/1.0

### PR DESCRIPTION
**Pull request by GearsAD**
Mostly this is a switch-out of Request.jl for HTTP.jl and some refactoring of the tests. Everything seems to work on my local environment, but could you check in yours just in case I missed something?

@kilminster FYI as we just saw that you have fork with 0.7 updates.

@dehann

### CODE REVIEW

**Comment by dehann** [comment]
Hi, please see the updated `.travis.yml` file pasted below.  I do not currently have access to @GearsAD 's repo fork to affect the fixes myself:

```travis
language: julia
julia:
  - 0.7
  - 1.0
  - nightly
notifications:
  email: false
## (tests will run but not make your overall status red)
matrix:
  allow_failures:
    - julia: 1.0
    - julia: nightly

before_install:
  - wget dist.neo4j.org/neo4j-community-3.3.5-unix.tar.gz
  - tar -xzf neo4j-community-3.3.5-unix.tar.gz
  - sed -i 's/#dbms.security.auth_enabled=false/dbms.security.auth_enabled=false/g' neo4j-community-3.3.5/conf/neo4j.conf
  - sed -i 's/dbms.security.auth_enabled=true/dbms.security.auth_enabled=false/g' neo4j-community-3.3.5/conf/neo4j.conf
  - neo4j-community-3.3.5/bin/neo4j start

script:
  - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
  - julia --check-bounds=yes -e 'using Pkg; Pkg.clone(pwd()); Pkg.test("Neo4j"; coverage=true)'

after_success:
  - julia -e 'using Pkg; cd(Pkg.dir("Neo4j")); Pkg.add("Coverage"); using Coverage; Codecov.submit(process_folder())'
```

**Comment by dehann** [comment]
Okay, got access and updated `.travis.yml`.

**Comment by dehann** [comment]
Hi @glesica , @StefanHaunsberger  , we updated code for Julia 0.7 / 1.0.  Could I maybe ask you to tag a new release of this repo please?  Looks like you might already be using [Julia's Attobot](https://github.com/attobot/attobot).   Thanks in advance!

File: README.md [hunk 1]
```diff
@@ -29,9 +36,9 @@ Rollbacks are also supported:
 rollback(tx)
 ```
 
-If the goal is to simply run a MATCH query and having the result in the form of a 
+If the goal is to simply run a MATCH query and having the result in the form of a
 `DataFrames.DataFrame` object, the `cypherQuery` function can be used.
-The `cypherQuery` implementation performs the query in a single trnsaction which 
+The `cypherQuery` implementation performs the query in a single trnsaction which
```
Comment (glesica) [inline README.md]: Can we fix "trnsaction" while we're fiddling with the readme?

File: README.md [hunk 2]
```diff
@@ -29,9 +36,9 @@ Rollbacks are also supported:
 rollback(tx)
 ```
 
-If the goal is to simply run a MATCH query and having the result in the form of a 
+If the goal is to simply run a MATCH query and having the result in the form of a
```
Comment (glesica) [inline README.md]: s/having the result/get the result/ maybe?

File: REQUIRE [hunk 3]
```diff
@@ -1,6 +1,6 @@
-julia 0.6
-Compat 0.7.13
+julia 0.7
 JSON
-Requests
+HTTP
 DataFrames
 Missings
+DocStringExtensions 0.5.0
```
Comment (glesica) [inline REQUIRE:6]: Fun!

File: README.md (hunk 1 above)
Comment (dehann) [reply README.md]: Sure!

File: src/Neo4j.jl [hunk 4]
```diff
@@ -42,7 +44,7 @@ struct Connection
    user::AbstractString #UTF8String
    password::AbstractString #UTF8String
 
-   Connection{T <: AbstractString}(host::T; port = DEFAULT_PORT, path = DEFAULT_URI, tls = false, user = "", password = "") = 
+   Connection(host::T; port = DEFAULT_PORT, path = DEFAULT_URI, tls = false, user = "", password = "") where {T <: AbstractString} =
```
Comment (glesica) [inline src/Neo4j.jl:47]: Oh I like the new syntax, like C#, easier to read IMHO.

File: README.md (hunk 2 above)
Comment (dehann) [reply README.md]: Alright, all done in c7dcc5c

**Comment by glesica** [comment]
Looks good, I'll merge and release as soon as CI finishes.

**Comment by glesica** [comment]
Should we move to 2.0.0 for this release? Looks like a few of the function signatures have changed and it appears we've dropped compatibility with <=0.6.0, correct?

**Comment by dehann** [comment]
Thank you very much!

**Comment by dehann** [comment]
Correct, dropped 0.6, and a few changes.  Either 0.x.0, or x.0.0.  Either is fine in this case.

**Comment by dehann** [comment]
Hi @glesica , sorry one more thing... Do you have [Attobot](https://github.com/attobot/attobot) setup on this repo?  I was not able to find the Neo v2.0.0 tag PR on JuliaLang/METADATA.jl.  Attobot is real easy to set up -- and in this case you could just create a new tag, something like v2.0.1 or redo v2.0.0?

**Comment by glesica** [comment]
Done.

**Comment by GearsAD** [comment]
Awesome, thanks :+1:

**Comment by dehann** [comment]
Awesome, thank you very much! Sorry to bother you on a weekend.

**Pull request merged**